app/main.py

```diff
@@ -40,10 +41,11 @@ async def lifespan(app: FastAPI, registry: svcs.Registry):
     root_path="/v1",
     debug=settings.debug,
     lifespan=lifespan,
+    dependencies=[Depends(get_current_system)],
```

Comment:
I'm not sure this optimization will survive the introduction of roles and operators 😢. Anyway we can leave it as you changed and we will see.